myosuite/envs/myo/myochallenge/chasetag_v0.py

```diff
@@ -374,6 +386,176 @@ def ncol(self):
         return self.hfield.ncol
 
 
+class RepellerChallengeOpponent(ChallengeOpponent):
+    # Repeller parameters
+    DIST_INFLUENCE = 3.5 # Distance of influence by the repeller
+    ETA = 20.0 # Scaling factor
+    MIN_SPAWN_DIST = 1.5
+    BOUND_RESOLUTIONS = [-8.7, 8.7, 25]
+
+    def __init__(self,
+                 sim,
+                 rng,
+                 probabilities: tuple[float],
+                 min_spawn_distance: float,
+                 chase_vel_range: tuple[float],
+                 random_vel_range: tuple[float],
+                 repeller_vel_range: tuple[float],
+        ):
+        self.dt = 0.01
+        self.sim = sim
+        self.rng = rng
+        self.opponent_probabilities = probabilities
+
+        self.min_spawn_distance = min_spawn_distance
+        self.noise_process = pink.ColoredNoiseProcess(beta=2, size=(2, 2000), scale=10, rng=rng)
```

Comment:
Is this deterministic given seed?

Reply:
yes, the `rng` is passed from the environment and in sync.